clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -2857,32 +2857,59 @@ static void handleWeakImportAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
 }
 
 // Checks correctness of mutual usage of different work_group_size attributes:
-// reqd_work_group_size, max_work_group_size. Values of reqd_work_group_size
-// arguments shall be equal or less than values coming from max_work_group_size.
+// reqd_work_group_size, max_work_group_size and max_global_work_dim.
+// Values of reqd_work_group_size arguments shall be equal or less than values
+// coming from max_work_group_size.
+// In case of 'max_global_work_dim' attribute equals to 0 we shall ensure
```

Comment:
```suggestion
// In case the value of 'max_global_work_dim' attribute equals to 0 we shall ensure
```

Reply:
Why not " 'max_global_work_dim' attribute equals to 1" instead? 0 means no execution, and then we can quite speed up the compilation time, FPGA resource and execution time. ;-)

Reply:
That is actually a catch

Reply:
@keryell Please resolve the conversation/approve if the updated changes are fine with you.

Reply:
I have a feeling, that Ronan will be OK with either way we go...

Reply:
Ah yes, with the documentation here I understand the goal and I was wrong. Dorry for the noise.
That is good for FPGA.